src/test/java/de/medizininformatikinitiative/torch/service/CrtdlValidatorServiceTest.java

```diff
@@ -70,19 +70,40 @@ void referenceWithoutLinkedGroups() throws ValidationException {
     }
 
     @Test
-    void validInput() throws ValidationException {
-        Crtdl crtdl = new Crtdl(node, new DataExtraction(List.of(patientGroup, new AttributeGroup("test", "https://www.medizininformatik-initiative.de/fhir/core/modul-labor/StructureDefinition/ObservationLab", List.of(), List.of(new Filter("token", "code", List.of(new Code("some-system", "some-code"))))))));
+    void validInput_withoutFilter() throws ValidationException {
+        Crtdl crtdl = new Crtdl(node, new DataExtraction(List.of(patientGroup, new AttributeGroup("test", "https://www.medizininformatik-initiative.de/fhir/core/modul-labor/StructureDefinition/ObservationLab", List.of(), List.of()))));
 
         var validatedCrtdl = validatorService.validate(crtdl);
 
         assertThat(validatedCrtdl).isNotNull();
-        assertThat(validatedCrtdl.dataExtraction().attributeGroups().getFirst().compiledFilter()).isNull();
-        assertThat(validatedCrtdl.dataExtraction().attributeGroups().getFirst().attributes()).isEqualTo(
+        assertThat(validatedCrtdl.dataExtraction().attributeGroups().get(0).compiledFilter()).isNull();
+        assertThat(validatedCrtdl.dataExtraction().attributeGroups().get(1).compiledFilter()).isNull();
+        assertThat(validatedCrtdl.dataExtraction().attributeGroups().get(0).attributes()).isEqualTo(
                 List.of(
                         new AnnotatedAttribute("Patient.id", "Patient.id", "Patient.id", false),
                         new AnnotatedAttribute("Patient.meta.profile", "Patient.meta.profile", "Patient.meta.profile", false)
                 ));
+        assertThat(validatorService.validate(crtdl).dataExtraction().attributeGroups().get(1).attributes()).isEqualTo(
+                List.of(
+                        new AnnotatedAttribute("Observation.id", "Observation.id", "Observation.id", false),
+                        new AnnotatedAttribute("Observation.meta.profile", "Observation.meta.profile", "Observation.meta.profile", false),
+                        new AnnotatedAttribute("Observation.subject", "Observation.subject", "Observation.subject", false, List.of("patientGroupId"))
+                ));
+    }
+    @Test
+    void validInput_withFilter() throws ValidationException {
```

Comment:
please use only the observation AG with filter here.

Reply:
Ok sorry, you are right. It won't work without patient AG. But I would not write asserts for that patient AG to keep the test focused on the actual AG with filter to test.